tests/linux_kernel/kmod/drgn_test.c

```diff
@@ -847,6 +848,42 @@ static void drgn_test_idr_exit(void)
 	idr_destroy(&drgn_test_idr_sparse);
 }
 
+// wait-queue
+static struct task_struct *drgn_test_waitq_kthread;
+static wait_queue_head_t drgn_test_waitq;
+static wait_queue_head_t drgn_test_empty_waitq;
+static bool exit_waitq_test;
+
+static int drgn_test_waitq_kthread_fn(void *arg)
+{
+	wait_event_interruptible(drgn_test_waitq, exit_waitq_test);
+	return 0;
+}
+
+static int drgn_test_waitq_init(void)
+{
+	drgn_test_waitq_kthread = kthread_create(drgn_test_waitq_kthread_fn, NULL,
+					   "drgn_test_waitq_kthread");
+	if (!drgn_test_waitq_kthread)
+		return -1;
+
+	init_waitqueue_head(&drgn_test_waitq);
+	init_waitqueue_head(&drgn_test_empty_waitq);
+
+	wake_up_process(drgn_test_waitq_kthread);
+	return 0;
+}
+
+static void drgn_test_waitq_exit(void)
+{
+	if (drgn_test_waitq_kthread) {
+		exit_waitq_test = true;
+		kthread_stop(drgn_test_kthread);
+		drgn_test_kthread = NULL;
```

Comment:
I think this was supposed to be:

```suggestion
		kthread_stop(drgn_test_waitq_kthread);
		drgn_test_waitq_kthread = NULL;
```